### PR TITLE
build(docs-infra): bump @angular-eslint packages to 12.5.0

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
update the four @angular-eslint packages from 12.4.1 to 12.5.0,
also add the caret to allow minor version updates

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

Simple update of aio @angular-eslint/* npm packages

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
As discussed here: https://github.com/angular/angular/pull/43218
(@IgorMinar , @gkalpak :slightly_smiling_face::+1: )

### CODE REVIEW

File: aio/package.json [hunk 1]
```diff
@@ -107,10 +107,10 @@
...
   "devDependencies": {
     "@angular-devkit/build-angular": "12.1.1",
-    "@angular-eslint/builder": "12.4.1",
-    "@angular-eslint/eslint-plugin": "12.4.1",
-    "@angular-eslint/eslint-plugin-template": "12.4.1",
-    "@angular-eslint/template-parser": "12.4.1",
+    "@angular-eslint/builder": "^12.5.0",
+    "@angular-eslint/eslint-plugin": "^12.5.0",
+    "@angular-eslint/eslint-plugin-template": "^12.5.0",
+    "@angular-eslint/template-parser": "^12.5.0",
```
Comment (dario-piotrowicz) [inline aio/package.json:113]: I took this chance to also add the caret, I hope it's fine, I think it makes sense here, we do have some dependencies with the caret and others without, not too sure if there is a rule in place for when to use it or not :sweat_smile:

Comment (gkalpak) [reply aio/package.json:113]: It shouldn't matter, because we pin to specific version via the lockfile anyway, so either way is fine.

**Review by gkalpak — APPROVED** [review]
✨

File: aio/package.json (hunk 1 above)
Comment (dario-piotrowicz) [reply aio/package.json:113]: ok cool, thanks! :slightly_smiling_face:

**Comment by alxhub** [comment]
@dario-piotrowicz this doesn't apply cleanly to 12.2.x. I'm going to merge it to `master` - please open a separate PR if it also needs to go into the patch branch.

**Pull request closed without merging**